mixins/form.jsx

```diff
@@ -411,7 +417,13 @@ module.exports = {
         submitting: true
       });
       submitProps = form.buildProps(props);
-      this.submit("/api/signup", submitProps, this.signupSuccess, this.signupError);
+      this.submit(url, submitProps, success, error);
     }
+  },
+  signup: function(validate, props) {
+    this.doSignup("/api/signup", validate, props, this.signupSuccess, this.signupError);
```

Comment:
I don't want to go digging in joi, I'm looking for something simpler. I'll take your first suggestion `/api/signup/basket` and the other one `/api/signup/mailchimp` if it gets an r+, or what I currently have up for review.

If we want to go further, I'm going to say we push it to another ticket as fiddling with joi isn't a blocker to shipping thunderbird, which has dragged on waaay too long now.